src/components/controls/km/km-validation-errors.vue

```diff
@@ -0,0 +1,107 @@
+    >
+      {{ t("valid") }}
+    </div>
+    <div v-if="!isAnalyzing && !isSaving">
+      <div
+        v-if="!hideErrors && errors && errors.length"
+        class="alert alert-warning"
+        role="alert"
+      >
+        <button
+          type="button"
+          @click="onButtonClose"
+          aria-label="Close"
+          class="position-absolute top-0 end-0 m-2 bg-transparent border-0 fs-5 text-secondary">
+          <span aria-hidden="true">&times;</span>
+        </button>
+        <strong>
+          {{ tc("contain", errors.length) }} {{ errors.length }} {{ tc("error", errors.length) }}
+        </strong>
+
+        <ul>
+          <li v-for="error in errors" :title="error.parameters" :key="error">
+            <span>{{ t(getTranslation(error)) }}</span>
+            <ul v-if="error.properties">
+              <li
+                v-for="property in error.properties"
+                :title="error.parameters"
+                :key="property"
+              >
+                <a
+                  class="text-decoration-none"
+                  href="#"
+                  @click.prevent="jumpTo($event, property.property)"
+                >
+                  {{ getLabel(property.property) }}
+                </a>
+              </li>
+            </ul>
+            <span v-if="!error.properties">
+              <a
+                class="text-decoration-none"
+                href="#"
+                @click.prevent="jumpTo($event, error.property)"
+              >
+                {{ getLabel(error.property) }}
+              </a>
+            </span>
+          </li>
+        </ul>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import {useAttrs, defineEmits, defineProps, computed, inject, ref} from "vue";
+import { useI18n } from '../../../services/composables/i18n'
+
+const { t, tc, locale } = useI18n();
+
+const props = defineProps({
+  errors: { type: Array, required: true },
+  isSaving: { type: Boolean, required: false },
+  isAnalyzing: { type: Boolean, required: false },
+  allowShowErrors: { type: Boolean, default: true },
+});
+const hideErrors = ref(false);
+const $emits = defineEmits(["onJumpTo"]);
+const showErrors = computed(() =>!hideErrors && props.errors && props.errors.length && props.allowShowErrors);
+const container = useAttrs().container ?? "body,html";
+const getLabel = inject("getLabel");
+
+function getTranslation(error) {
+  if (error?.code === null) return "unknown";
+  if (error?.code == "Error.Mandatory") return "mandatory";
+  if (error?.code == "Error.MandatoryAnyOf") return "mandatoryAnyOf";
+  if (error?.code == "Error.InvalidValue") return "invalidValue";
+  if (error?.code == "Error.InvalidProperty") return "invalidProperty";
+  if (error?.code == "Error.UnspecifiedLocale") return "unspecifiedLocale";
+  if (error?.code == "Error.UnexpectedTerm") return "unexpectedTerm";
+  if (error?.code == "Error.InvalidType") return "invalidType";
+
+  return error?.code;
+}
```

Comment:
we shoudl just move this to translation file instead of a fn

Reply:
moved translation to "json" file and used i18n function